### PR TITLE
cloudns: fix TTL and status validation

### PR DESCRIPTION
**Pull request by ldez**
~~If I understand correctly the "not clear" documentation of ClouDNS, the List records operation returns an array: https://www.cloudns.net/wiki/article/57/~~

related to https://github.com/containous/traefik/issues/4772

ping @tzakrajs

### CODE REVIEW

**Comment by steve-todorov** [comment]
@ldez Thank you for looking into this! Here's a JSON response from the API:

```
$ curl -s -H accept:application/json "https://api.cloudns.net/dns/records.json?auth-id=my-id&auth-password=a-password&domain-name=domain.com&type=txt" | jq
{
  "5769228": {
    "id": "5769228",
    "type": "TXT",
    "host": "",
    "record": "v=spf1 +a +mx include:_spf.google.com -all",
    "failover": "0",
    "ttl": "3600",
    "status": 1
  },
  "181805209": {
    "id": "181805209",
    "type": "TXT",
    "host": "_github-challenge",
    "record": "b66b8324b5",
    "failover": "0",
    "ttl": "300",
    "status": 1
  }
}

```

Please let me know if I could help with anything more.

**Comment by ldez** [comment]
@steve-todorov  thank you, so it's not an array but a map.

**Comment by ldez** [comment]
If fact the API returns `[]` when there is no record and an object when there are records.

Very strange API, I updated the behavior.

File: providers/dns/cloudns/internal/client.go [hunk 1]
```diff
@@ -130,7 +135,7 @@ func (c *Client) FindTxtRecord(zoneName, fqdn string) (*TXTRecord, error) {
 		}
 	}
 
-	return nil, fmt.Errorf("ClouDNS: no existing record found for %q", fqdn)
+	return nil, nil
```
Comment (wyattjoh) [inline providers/dns/cloudns/internal/client.go]: Why aren't you returning an error here anymore?

Comment (ldez) [reply providers/dns/cloudns/internal/client.go]: I manage the `nil` inside the provider, seems like a better way to manage this part.

Comment (ldez) [reply providers/dns/cloudns/internal/client.go]: the method `FindTxtRecord` is only use for the clean up.

**Pull request merged**